sdk/aqueduct/integrations/mongodb_integration.py

```diff
@@ -0,0 +1,171 @@
+            if artf.type != ArtifactType.STRING:
+                raise InvalidUserArgumentException(
+                    "The parameter `%s` must be defined as a string. Instead, got type %s"
+                    % (artf.name, artf.type)
+                )
+        param_artf_ids = [artf.id for artf in param_artifacts]
+        op_id = generate_uuid()
+        output_artf_id = generate_uuid()
+        apply_deltas_to_dag(
+            self._dag,
+            deltas=[
+                AddOrReplaceOperatorDelta(
+                    op=Operator(
+                        id=op_id,
+                        name=op_name,
+                        description=description,
+                        spec=OperatorSpec(
+                            extract=ExtractSpec(
+                                service=self._metadata.service,
+                                integration_id=self._metadata.id,
+                                parameters=mongo_extract_params,
+                            )
+                        ),
+                        inputs=param_artf_ids,
+                        outputs=[output_artf_id],
+                    ),
+                    output_artifacts=[
+                        ArtifactMetadata(
+                            id=output_artf_id,
+                            name=artifact_name_from_op_name(op_name),
+                            type=ArtifactType.TABLE,
+                        ),
+                    ],
+                ),
+            ],
+        )
+
+        if execution_mode == ExecutionMode.EAGER:
+            # Issue preview request since this is an eager execution.
+            artifact = artifact_utils.preview_artifact(self._dag, output_artf_id)
+            assert isinstance(artifact, TableArtifact)
+            return artifact
+        else:
+            # We are in lazy mode.
+            return TableArtifact(self._dag, output_artf_id)
+
+    def config(self, update_mode: LoadUpdateMode) -> SaveConfig:
+        return SaveConfig(
+            integration_info=self._metadata,
+            parameters=RelationalDBLoadParams(table=self._collection_name, update_mode=update_mode),
+        )
+
+
+class MongoDbIntegration(Integration):
+    """
+    Class for MongoDB integration. This works similar to mongo's `Database` object:
+
+    mongo_integration = client.integration("my_integration_name")
+    my_table_artifact = mongo_integration.my_table.find({})
+    """
+
+    def __init__(self, dag: DAG, metadata: IntegrationInfo):
+        self._dag = dag
+        self._metadata = metadata
+
+    def __getattr__(self, name: str) -> Any:
+        """
+        Overrided `__get_attr__` allows caller to access integration by
+        arbitrary attribute with similar experience as mongo API:
+
+        mongo_integration = client.integration("my_integration_name")
+        my_table_artifact = mongo_integration.my_table.find({})
+
+        """
+        return MongoDbCollectionIntegration(self._dag, self._metadata, name)
```

Comment:
this is neat! It makes me wonder if we can do a similar thing when previewing operators with custom backends:
```
@op
def foo(sql):
    ...

foo.my_lambda_integration(input_artf)
```

Reply:
Might be a bit too much magic though, now that I think about it, but it would be consistent with this. I'm curious if there was a reason you went for this proposal instead of:
```
mongo = client.integration(name="mongo_sample")
mongo.collection("reviews").find(...)
```
Since ^ actually seems more understandable to me.

Reply:
This is how you access collection using a mongo's db object: https://www.mongodb.com/docs/manual/tutorial/query-documents/ so the overall experience is the same.

I don't think we should generalize this to other use cases as I myself find it confusing to learn. I also find what you recommend more intuitive. So it's more a question of consistent with Mongo (who's not consistent with other relational DB :) ) or consistent with us. I slightly prefer the former as we need to be mongo-consistent in the `.find()` part anyway

Reply:
@vsreekanti would like to get your thoughts as well ^^

Reply:
I had the same exact intuition - consistent for us vs consistent with Mongo. I actually have a slight preference for the latter, since I think objectively the pattern is cool but weird and I doubt its natural enough for users to try and generalize this pattern to other APIs. Also, I value keeping our API more consistent. But yea lets just chat at the sync.